pkg/controller/operconfig/migration.go

```diff
@@ -97,3 +112,313 @@ func convertEgressFirewallToEgressNetworkPolicy(ctx context.Context, client cnoc
+			if err != nil {
+				return err
+			}
+
+			// now remove egressIP config from netnamespace
+			nns.Object["egressIPs"] = nil
+			_, err = client.Default().Dynamic().Resource(gvrNetnamespace).Update(ctx, &nns, metav1.UpdateOptions{})
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil // success
+}
+
+func convertOvnEgressIpToSdnEgressIp(ctx context.Context, client cnoclient.Client) error {
+	egressIpRollbackReady, err := sdnEgressIpResourcesReady(ctx, client)
+	if !egressIpRollbackReady {
+		return nil // wait for all SDN egressIP resources to be created before rollback
+	} else if err != nil {
+		return err
+	}
+
+	// 1. query for egressips
+	eipList, err := client.Default().Dynamic().Resource(gvrEgressIp).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	// 2. query for nodes
+	nodeList, err := client.Default().Kubernetes().CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	// 3. iterate through egressIP objects and generate netnamespaces
+	//    - value of `namespaceSelector` will determine netnamespace's name
+	//    - egressIP list will be copied to egressIPs field for netnamespace
+	for _, eip := range eipList.Items {
+		// translate egressIPs to SDN netnamespaces
+		// - delete cloudprivateipconfig if it exists
+		// - create egressip to netnamespace resources for sdn
+
+		egressIps := eip.Object["spec"].(map[string]interface{})["egressIPs"].([]interface{})
+
+		for _, egressIp := range egressIps {
+			egressIpStr := fmt.Sprintf("%v", egressIp)
+			err = client.Default().Dynamic().Resource(gvrCloudPrivateIPConfig).Delete(ctx, egressIpStr, metav1.DeleteOptions{})
+			if err != nil {
+				return err
+			}
+		}
+
+		eipNamespace := eip.Object["spec"].(map[string]interface{})["namespaceSelector"].(map[string]interface{})["matchLabels"].(map[string]interface{})["kubernetes.io/metadata.name"]
+		eipNamespaceStr := fmt.Sprintf("%v", eipNamespace)
+		nns, err := client.Default().Dynamic().Resource(gvrNetnamespace).Get(ctx, eipNamespaceStr, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		nnsResourceVersion := nns.Object["metadata"].(map[string]interface{})["resourceVersion"]
+		nnsNetId := nns.Object["netid"]
+		netName := eip.Object["spec"].(map[string]interface{})["namespaceSelector"].(map[string]interface{})["matchLabels"].(map[string]interface{})["kubernetes.io/metadata.name"]
+
+		netnamespace := &uns.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "network.openshift.io/v1",
+				"kind":       "NetNamespace",
+				"metadata": map[string]interface{}{
+					"name":            netName,
+					"resourceVersion": nnsResourceVersion,
+				},
+				"netid":     nnsNetId,
+				"netname":   netName,
+				"egressIPs": egressIps,
+			},
+		}
+		_, err = client.Default().Dynamic().Resource(gvrNetnamespace).Update(ctx, netnamespace, metav1.UpdateOptions{})
```

Comment:
Shall we use `Patch` instead of `Update` here? Also, we should check if the netnamespace object is already in the desired state before updating it.

Reply:
I don't see a benefit of switching to `Patch`, `Update` works fine in my tests. Also when performing a rollback, the netnamespace object will not be in the desired state (namely, that egressIPs are configured) because when rolling back, SDN creates the netnamespaces/hostsubnets with no additional configuration. Therefore the `egressCIDRs/egressIPs` fields are left empty and updating is necessary.

Reply:
> For example, if multicast migration creates an annotation during rollback, it will be forcefully overwritten by an Update call. Thank you for this suggestion.

This is already resolved, but just for book-keeping, I think the highlighted comment I had made earlier was incorrect. The reason things were being "forcefully overwritten" by `Update` is because I was defining an entirely new object and sending that into the `Update`. The proper way is to perform a `Get`, modifying any necessary fields (in this case the egressIPs field) and then run an `Update` on the object, that way the original config is preserved, alongside any updates you provide. Due to some comments by @jcaamano it appears using `Update` will be safer than `ApplyObject` (see discussion in later review comments)